pkg/controller/volume/expand/expand_controller.go

```diff
@@ -218,14 +247,35 @@ func (expc *expandController) syncHandler(key string) error {
 		}
 		expc.recorder.Event(pvc, eventType, events.ExternalExpanding, fmt.Sprintf("Ignoring the PVC: %v.", msg))
 		klog.Infof("Ignoring the PVC %q (uid: %q) : %v.", util.GetPersistentVolumeClaimQualifiedName(pvc), pvc.UID, msg)
-		return err
+		// If we are expecting that an external plugin will handle resizing this volume then
+		// is no point in requeuing this PVC.
+		return nil
+	}
+
+	if volumePlugin.IsMigratedToCSI() {
+		msg := fmt.Sprintf("CSI migration enabled for %s; waiting for external resizer to expand the pvc", volumeResizerName)
+		expc.recorder.Event(pvc, v1.EventTypeNormal, events.ExternalExpanding, msg)
+		csiResizerName, err := csitranslation.GetCSINameFromInTreeName(class.Provisioner)
+		if err != nil {
+			errorMsg := fmt.Sprintf("error getting CSI driver name for pvc %s, with error %v", util.ClaimToClaimKey(pvc), err)
+			expc.recorder.Event(pvc, v1.EventTypeWarning, events.ExternalExpanding, errorMsg)
```

Comment:
but the logs will only be accessible to admin and I think we should surface any errors associated with resizing in PVC itself.

Reply:
I think the idea was to return error to caller and the caller records event, instead of individual event creation in this function.

I can see there are paths that return `err` without creating event, not sure if they're worth creating an event or not.

Reply:
makes sense. I was under the impression any error was also logged as event. If that is not the case please ignore this comment